### PR TITLE
[LOR-221] - create layout summary screen

### PR DESCRIPTION
**Pull request by rodrigoliveirac**
## What's this PR doing?

- [x] Create a layout for the SummaryScreen with a toolbar component.
- [x] Add SummaryView to the NavigationStack.
- [x] Disable user navigation forward if no task is selected.
- [x] Disable user navigation forward if neither Shift nor Frequency is selected.
- [x] Retrieve all LocbookTask data and display it on the SummaryScreen.
- [x] Enhance the Shift Tab Row component by implementing dynamic changes to the background color and foreground color based on the selected shift option.

## Preview
LOR-221
> Note: The user selected the afternoon shift.

Screenshot 2024-02-07 at 11 32 37

> Note: When the user selected the morning shift.

Screenshot 2024-02-07 at 11 33 00

> Note: When the user selected the night shift.

[LOR-221]: https://lorybluapp.atlassian.net/browse/LOR-221?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: loryblu/loryblu/Feature/Locbook/View/FrequencyRotineView.swift [hunk 1]
```diff
@@ -81,8 +81,16 @@ struct FrequencyRotineView: View {
             Spacer()
 
             LBButton(title: LBStrings.General.confirm) {
+                
                 formConfig.task.frequency = formConfig.frequency
-                props.onNext?(formConfig.task)
+                
+                if formConfig.task.shift == nil {
```
Comment (rodrigoliveirac) [inline loryblu/loryblu/Feature/Locbook/View/FrequencyRotineView.swift:87]: > If the user does not select a shift, set a morning shift by default.

File: loryblu/loryblu/Components/View/ShiftsView/LBShiftItemsComponent.swift [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+//
+//  Created by Rodrigo Cavalcante on 05/02/24.
+//
+
+import SwiftUI
+
+struct ShiftItem {
+    var name: String
+    var icon: String
+    var backgroundColor: Color
```
Comment (sucriss2) [inline loryblu/loryblu/Components/View/ShiftsView/LBShiftItemsComponent.swift:13]: Como sugestao , podemos usar aqui o nosso tipo Color. LBColor

File: loryblu/loryblu/Components/View/ShiftsView/LBShiftItemsComponent.swift [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    var letterColor: Color
```
Comment (sucriss2) [inline loryblu/loryblu/Components/View/ShiftsView/LBShiftItemsComponent.swift:14]: Como sugestao , podemos usar aqui o nosso tipo Color. LBColor

File: loryblu/loryblu/Components/View/ShiftsView/LBShiftItemsComponent.swift [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                        backgroundColor: shift.backgroundColor,
+                        letterColor: shift.letterColor)
+                } else {
+                    Text(shift.name)
+                        .font(LBFont.bodyLarge)
+                        .foregroundColor(.gray)
+                        .frame(maxWidth: .infinity,alignment: .center).padding(2)
+                }
+            }
+        }.background(LBColor.textfield).cornerRadius(8).frame(maxWidth:.infinity,alignment: .center)
```
Comment (sucriss2) [inline loryblu/loryblu/Components/View/ShiftsView/LBShiftItemsComponent.swift:38]:
```
 }
.background(LBColor.textfield)
.cornerRadius(8)
.frame(
    maxWidth:.infinity,
    alignment: .center
)
```

File: loryblu/loryblu/Components/View/ShiftsView/LBSelectedItemShift.swift [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+//  Created by Rodrigo Cavalcante on 05/02/24.
+//
+
+import SwiftUI
+
+struct LBSelectedItemShift: View {
+    
+    let shiftName: String
+    let iconValue: String
+    let backgroundColor: Color
```
[the remark this answers is not in the document]
Comment (rodrigoliveirac) [reply loryblu/loryblu/Components/View/ShiftsView/LBSelectedItemShift.swift:14]: Ficaria inviável, pois o parâmetro do método `.background()` só pode receber uma Color.

Por exemplo, não se pode fazer isso:

let letterColor:LBColor
    
    var body: some View {
        HStack {
           (...)
        }.background(letterColor)
    }

O Problema do LBColor é que suas propriedades são uma Color, se fossem Strings seria o mais ideal, pois daí ficaria assim:

let letterColor: String
    
    var body: some View {
        HStack {
           (...)
        }.background(Color(letterColor))
    }

**Review by Rodrigo-Eduardo-Silva — APPROVED** [review]

**Review by sucriss2 — APPROVED** [review]

**Pull request merged**